src/enochecker/enochecker.py

```diff
@@ -747,13 +748,21 @@ def connect(
         :param host: the host to connect to (defaults to self.address)
         :param port: the port to connect to (defaults to self.port)
         :param timeout: timeout on connection (defaults to self.timeout)
+        :param retries: how often establishing the connection should be retries
         :return: A connected Telnet instance
         """
         if timeout:
             timeout_fun: Callable[[], int] = lambda: cast(int, timeout)
         else:
             timeout = self.time_remaining // 2
-            timeout_fun = lambda: self.time_remaining // 2
+
+            # PEP8 wants this to be a function, not a lambda...
+            def timeout_fun():
+                return self.time_remaining // 2
+
+        if retries < 1:
+            self.info(f"Max connection retries for {host}:{port} reached, giving up.")
+            raise OfflineException(f"Connection to service failed.")
```

Comment:
If I say "don't retry this" this won't do anything instead of "don't retry this"